### ISSUE TITLE
Saved logins for subdomains not autofilling on Android Oreo

### ISSUE THREAD

**Issue by Techn1x** [bitwarden/mobile#321]
I have a user/pass login saved, with URI for the login set to `some-dev.testing.fakedomain.com:8080`

Then, when I access the site
* in **Chrome**, I right click the field > Bitwarden > it offers the correct autofill options for that whole domain, so this works
* in **Android**, the notification to autofill appears, I tap it, then it takes me to "items for fakedomain.com" - but there are no items in there.

I have checked that Android has definitely synced the login, and it appears in the vault.
Am I doing something wrong? Or does the app not yet recognize subdomains?

OS: Android Oreo 8.0.0
Device: OnePlus 3

**Comment by kspearrin** [comment]
What URI match setting are you using?

**Comment by Techn1x** [comment]
Didn't even see that setting! :D (I'm new to Bitwarden)

It's set to "Default Match Detection", which after reading the Match Detection documentation, seems to default to "Base Domain". I am guessing this is still an issue because the match behaviour between Chrome and Android probably shouldn't be different.

After some more investigation I found the following;
* Removing the port number at the end of the URI makes no difference
* Adding `http://` to the start of the URI fixes the match detection on Android (this is fine since the website is HTTP only - perhaps this has something to do with it?)

So basically, on Android:
`some-dev.testing.fakedomain.com:8080` doesn't work
`http://some-dev.testing.fakedomain.com:8080` does work

**Comment by eliykat** [comment]
Thanks for reporting this issue. Autofill problems can affect different sites, apps, or devices, and we’re working on improving this feature. To help us track and analyze affected sites, please lodge a report using the Google Form mentioned in this issue: #1389. Please also direct any discussion or questions to that issue. This issue will now be closed.
